cmd/restic/cmd_recover.go

```diff
@@ -24,15 +25,29 @@ Exit status is 0 if the command was successful, and non-zero if there was any er
 `,
 	DisableAutoGenTag: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return runRecover(globalOptions)
+		return runRecover(globalOptions, recoverOptions)
 	},
 }
 
+type RecoverOptions struct {
+	All     bool
+	Combine bool
+	DryRun  bool
+	Tags    []string
+}
+
+var recoverOptions RecoverOptions
+
 func init() {
 	cmdRoot.AddCommand(cmdRecover)
+	recoverFlags := cmdRecover.Flags()
+	recoverFlags.BoolVar(&recoverOptions.All, "all", false, "add all directories, not only unreferenced ones")
```

Comment:
Do you see a use case where one would want to add root directories already covered by existing snapshots?

Reply:
No, actually not :wink: 
The current version effectively is `--all=true` and therefore I added this option to preserve the current behavior. I can remove this option and always remove directories covered by existing snapshots.

Reply:
I removed the `--all` option.